### DISCUSSION TITLE
Why does backpack_url() return http:// URL while APP_URL contains https:// ?

### DISCUSSION THREAD

**Discussion by gabrielfigueroa** [Laravel-Backpack/community-forum#716] [Q&A (Help)] [answered]
I am getting errors loading assets as a result of this. I refreshed cache with:

php artisan config:cache

and

php artisan route:cache

Same result.

**Comment by MaraIGI** [comment]
Hi Gabriel,

is it truly `backpack_url()` or Laravel `url()` error?

`backpack_url()` uses Laravel `url()` function, and if your are not passing `$secure param set to false` to the `backpack_url()` function (see backpack_url params: `function backpack_url($path = null, $parameters = [], $secure = null)` the problem should be in Laravel maybe?

Also check your `app/config.php` file if you accidently didn't rewrite `url` setting from default which is:
`'url' => env('APP_URL', 'http://localhost'),`, then it couldn't set it from `.env` file which you mentioned (APP_URL param which is normally in .env file)

-Mark

**Comment by pxpm** [comment, marked as answer]
Hey @gabrielfigueroa thanks for the question and thanks for the answer @MaraIGI 

It's like @MaraIGI said, we use the URL helper under the hood, we just add the "admin prefix segment" to the url. 

Please ensure your .env is correct, you can also enforce https in your `RouteServiceProvider`: 
```php
if($this->app->environment('production')) {
    \URL::forceScheme('https');
}
```

Hope that helps. 

Cheers
